lib/ddtrace/sampler.rb

```diff
@@ -42,8 +42,58 @@ def sample_rate=(sample_rate)
     end
 
     def sample(span)
-      span.sampled = ((span.trace_id * KNUTH_FACTOR) % Datadog::Span::MAX_ID) <= @sampling_id_threshold
       span.set_metric(SAMPLE_RATE_METRIC_KEY, @sample_rate)
+      span.sampled = ((span.trace_id * KNUTH_FACTOR) % Datadog::Span::MAX_ID) <= @sampling_id_threshold
+    end
+  end
+
+  # \RateByServiceSampler samples different services at different rates
+  class RateByServiceSampler < Sampler
+    def initialize(rate = 1.0, opts = {})
+      @env = opts.fetch(:env, Datadog.tracer.tags[:env])
+      @mutex = Mutex.new
+      @fallback = RateSampler.new(rate)
```

Comment:
It's perfectly fine to have a fallback and identify it as such, however keep in mind that the agent will report a value for this (typically with a key `service:,env:`) and this should be used as the library can't really have a clue of this default "fallback" value unless it queries the agent. Eg: https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/sampler.py#L90

Reply:
Oh, I see! OK, I think I'll have to change this. Thanks for spotting that 👍